### PR TITLE
add a decorator to mark a function as a Servable

### PR DESCRIPTION
**Pull request by seanshi-scale**
upd: this is deprecated, we're not doing decorators

Intended usage:

User does
```
@step_decorator
def some_stage(input1, input2):
    ...

...

def orchestrator(input1):
    result = some_stage(input1, 1)
```
or
```
@step_decorator_class
class SomeClass:
    ...
    def __call__(self, input2):
        ...

def orchestrator(input2):
    class_thing = SomeClass(something)
    res = class_thing(input2)
```
in their code to be run by us, and we turn `some_stage` into a function that can either make a network request, or run code locally. Note: the decorator does not need to be attached to the root stage, so users who want single-stage pipelines, or using legacy bundles, can still use the bundles.

### CODE REVIEW

**Comment by syandroo** [comment]
looks good overall but is there a reason we need both the class and the fn version of this?

File: launch/servable.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    # Identifiers will just be the name of the function. Must be unique. TODO enforce this uniqueness
+    fn_name = fn.__name__
+
+    def modified_fn(*args, **kwargs):
+        return make_request(fn_name, fn, args, kwargs)
+
+    return modified_fn
+
+
+def step_decorator_class(class_obj):
```
Comment (syandroo) [inline launch/servable.py:32]: any reason we need both the fn version and the class?

**Review by syandroo — APPROVED** [review]

File: launch/servable.py (hunk 1 above)
Comment (seanshi-scale) [reply launch/servable.py:32]: tbh I think we could have just checked whether something was a function or a class

having said that, I think the pr is not what we're gonna end up going with (I think we're gonna have people subclass some `Servable` class), I should have closed this pr lol mb

**Comment by seanshi-scale** [comment]
We're probably not going with this decorator method of marking/modifying user code. Closing.

**Pull request closed without merging**